go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/vmware-tanzu/load-balancer-operator-for-kubernetes
 
-go 1.22
+go 1.23
```

Comment:
shall we also update the github workflows on this branch to use go 1.23 in tests etc.? i believe `actions.yaml` and `push.yaml` both specify a go version

Reply:
also just a note for future dep bumps, using the `1.23.0` version (with the patch release here) prevents any issues with toolchain bumps etc., see: https://go.dev/doc/toolchain#version

Reply:
based on the issue description we might want to set a minimum of 1.23.1?

Reply:
Oh, I thought it's same when pulling image, 1.23 will pull the lastest patch, turns out not
```
For example, 1.21 < 1.21rc1 < 1.21rc2 < 1.21.0 < 1.21.1 < 1.21.2.
```